cmd/protoc-gen-goclay/genhandler/template.go

```diff
@@ -203,17 +203,30 @@ func addValueTyped(f *descriptor.Field) string {
 		if valueFormatter != "" {
 			return fmt.Sprintf(valueFormatter, getter)
 		}
+
+		if f.GetProto3Optional() {
+			getter = fmt.Sprintf("*%s", getter)
+		}
+
 		return fmt.Sprintf(`fmt.Sprintf("%s", %s)`, valueVerb, getter)
 	}
 
-	if !isRepeated {
+	if isRepeated {
```

Comment:
It is not possible to have both optional and repeated for the same field, right?
It does not make much sense to me, but is it possible from the protobuf perspective?

Reply:
To the best of my knowledge, repeated fields are inherently optional. If we try to put both optional and repeated on the same field, the parser will raise an error (did a quick experiment btw)

Reply:
Also, an answer on Stack Overflow -- https://stackoverflow.com/a/25637833